bitcoin/src/blockdata/transaction.rs

```diff
@@ -786,12 +786,10 @@ pub fn effective_value(
 ) -> NumOpResult<SignedAmount> {
     let weight = input_weight_prediction.total_weight();
 
-    let fee = match fee_rate.to_fee(weight) {
-        NumOpResult::Valid(x) => x.to_signed(),
-        NumOpResult::Error(e) => return NumOpResult::Error(e)
-    };
-
-    value.to_signed() - fee     // Cannot overflow.
+    fee_rate
+        .to_fee(weight)
+        .map(|fee| fee.to_signed())
```

Comment:
In eb684bbe54310c3db450972b4e22ca6e1108bab8:

Can you just use `.map(Amount::to_signed)`?

Reply:
done